truth_set/java/structure/organisation/move-functionality/UserService.java

```diff
@@ -0,0 +1,48 @@
+import java.util.List;
+import java.util.ArrayList;
+
+class Order {
+    private int id;
+    private String product;
+    private int quantity;
+
+    public Order(int id, String product, int quantity) {
+        this.id = id;
+        this.product = product;
+        this.quantity = quantity;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getProduct() {
+        return product;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+}
+
+class OrderService {
+    private List<Order> orders;
+
+    public OrderService() {
+        this.orders = new ArrayList<>();
+    }
+
+    public void placeOrder(Order order) {
+        orders.add(order);
+        notify("Order placed: " + order.getId());
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider separating order management concerns</b></div>
<div id="fix">

Consider extracting order management logic into a dedicated service class. The `placeOrder` method could be part of an `OrderManagementService` to better separate concerns.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -35,4 +35,4 @@
  public void placeOrder(Order order) {
      orderManagementService.addOrder(order);
      notificationService.notify("Order placed: " + order.getId());
  }

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/8#issuecomment-2561208575>#e3503f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged